book/src/snapshot-verification.md

```diff
@@ -0,0 +1,34 @@
+# Verifying Snapshots and Account state proof
+
+A snapshot includes the account state for each live account in the system at a given slot in a bundle of append vecs.
+
+To verify the snapshot, we propose the following:
+
+On account store hash the following data:
+* Account owner
+* Account data
+* Account pubkey
+* Account lamports balance
+* Fork the account is stored on
+
+Use this resulting hash value as input to an expansion function which expands the hash value into an image value.
+The function will create a 440 byte block of data where the first 32 bytes are the hash value, and the next 440 - 32 bytes
+are generated from a Chacha RNG with the hash as the seed.
+
+The account images are then combined with xor. The previous account value will be xored into the state and the new
+account value also xored into the state.
+
+Voting and sysvar hash values occur with the hash of the resulting full image value.
+
+The resulting value can be verified by a validator to be the result of xoring all current account states together.
+
+An attack on the xor state could be made to influence its value:
+
+Thus the 440 byte image size comes from this paper, avoiding xor collision with 0 (or thus any other given bit pattern):
+https://link.springer.com/content/pdf/10.1007%2F3-540-45708-9_19.pdf
+
+The math provides 128 bit security in this case:
+O(k · 2^(n/(1+lg(k)))
```

Comment:
Can you check the markdown preview? Probably want to wrap all these lines with triple backticks.